src/compile/data/assemble.ts

```diff
@@ -307,8 +312,8 @@ export function assembleFacetData(root: FacetNode): VgData[] {
  * @param  data array
  * @return modified data array
  */
-export function assembleData(dataCompomponent: DataComponent): VgData[] {
-  let roots: SourceNode[] = vals(dataCompomponent.sources);
+export function assembleData(model: Model, dataComponent: DataComponent): VgData[] {
```

Comment:
Can we pass just the selection component? 

If not, I'd rather pass the whole model to this method.